### PR TITLE
Feat/add pymotorcad methods for transient

### PR DESCRIPTION
**Pull request by DragosVMosteanu**
added thermal solution type to load_results and save_results
also added input check and version check for thermal for backwards compatibility
@JackB-Ansys could you please check this is working as intended?

I wasn't sure how to add the checks so let me know if there's a better way to do it
How can you test if it's working with thermal without having 25R1?

### CODE REVIEW

**Comment by jgsdavies** [comment]
Need to add unit testing.
See ```test_save_load_results()``` in ''test_general.py''' 

This needs re-adding with thermal results too. Let me know if you want any more info with this.

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 1]
```diff
@@ -356,31 +357,45 @@ def save_to_file(self, mot_file):
...
+            Soultion type, which must be ``"EMagnetic" or "Thermal"``.
         """
         method = "SaveResults"
         params = [solution_type]
+
+        if solution_type == "Thermal":
+            self.connection.ensure_version_at_least("2025.0")
+        elif solution_type != "EMagnetic":
+            raise MotorCADError(
+                "SaveResults not available for this solution type: " + solution_type
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:376]: Looks good thanks. Maybe worth adding the available types here too.

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 2]
```diff
@@ -356,31 +357,45 @@ def save_to_file(self, mot_file):
...
         Parameters
         ----------
         solution_type : str
-            Soultion type, which must be ``"EMagnetic"``.
+            Soultion type, which must be ``"EMagnetic" or "Thermal"``.
         """
         method = "SaveResults"
         params = [solution_type]
+
+        if solution_type == "Thermal":
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:372]: I would make sure this is case insensitive.

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 3]
```diff
@@ -356,31 +357,45 @@ def save_to_file(self, mot_file):
...
         Parameters
         ----------
         solution_type : str
-            Soultion type, which must be ``"EMagnetic"``.
+            Soultion type, which must be ``"EMagnetic" or "Thermal"``.
         """
         method = "LoadResults"
         params = [solution_type]
+
+        if solution_type == "Thermal":
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:393]: See save _results

**Review by jgsdavies — CHANGES_REQUESTED** [review]

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 4]
```diff
@@ -356,31 +357,45 @@ def save_to_file(self, mot_file):
...
+            self.connection.ensure_version_at_least("2025.0")
+        elif solution_type != "EMagnetic":
+            raise MotorCADError(
+                "LoadResults not available for this solution type: " + solution_type
```
Comment (PipKat) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:397]:
```suggestion
                "Load results are not available for this solution type: " + solution_type
```
Message strings should be in plain English if possible and have a verb, which was missing here.

Comment (DragosVMosteanu) [reply src/ansys/motorcad/core/methods/rpc_methods_general.py]: changed strings to have verb and plain english

File: src/ansys/motorcad/core/methods/rpc_methods_general.py (hunk 2 above)
Comment (DragosVMosteanu) [reply src/ansys/motorcad/core/methods/rpc_methods_general.py]: should now be case insensitive

File: src/ansys/motorcad/core/methods/rpc_methods_general.py (hunk 3 above)
Comment (DragosVMosteanu) [reply src/ansys/motorcad/core/methods/rpc_methods_general.py]: should now be case insensitive

File: src/ansys/motorcad/core/methods/rpc_methods_general.py (hunk 1 above)
Comment (DragosVMosteanu) [reply src/ansys/motorcad/core/methods/rpc_methods_general.py]: added new line to say available types

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 5]
```diff
@@ -356,31 +357,49 @@ def save_to_file(self, mot_file):
         return self.connection.send_and_receive(method, params)
 
     def save_results(self, solution_type):
-        """Save the output results from an ``"EMagnetic"`` solution.
+        """Save the output results from an ``"EMagnetic" or "Thermal"`` solution.
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:360]:
```suggestion
        """Save the output results from an ``"EMagnetic"`` or ``"Thermal"`` solution.
```

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 6]
```diff
@@ -356,31 +357,49 @@ def save_to_file(self, mot_file):
...
 
-        This method supports only ``"EMagnetic"`` solutions.
+        This method supports only ``"EMagnetic" or "Thermal"`` solutions.
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:362]:
```suggestion
        This method supports only ``"EMagnetic"`` or ``"Thermal"`` solutions.
```

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 7]
```diff
@@ -356,31 +357,49 @@ def save_to_file(self, mot_file):
...
 
         Parameters
         ----------
         solution_type : str
-            Soultion type, which must be ``"EMagnetic"``.
+            Solution type, which must be ``"EMagnetic" or "Thermal"``.
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:367]:
```suggestion
            Solution type, which must be ``"EMagnetic"`` or ``"Thermal"``.
```

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 8]
```diff
@@ -356,31 +357,49 @@ def save_to_file(self, mot_file):
...
+            raise MotorCADError(
+                "Save results are not available for this solution type: "
+                + solution_type
+                + "\nAvailable solution types are: Thermal, EMagnetic"
+            )
         return self.connection.send_and_receive(method, params)
 
     def load_results(self, solution_type):
-        """Load the output results from an ``"EMagnetic"`` solution.
+        """Load the output results from an ``"EMagnetic" or "Thermal"`` solution.
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:383]:
```suggestion
        """Load the output results from an ``"EMagnetic"`` or ``"Thermal"`` solution.
```

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 9]
```diff
@@ -356,31 +357,49 @@ def save_to_file(self, mot_file):
...
 
-        This method supports only ``"EMagnetic"`` solution.
+        This method supports only ``"EMagnetic" or "Thermal"`` solution.
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:385]:
```suggestion
        This method supports only ``"EMagnetic"`` or ``"Thermal"`` solution.
```

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 10]
```diff
@@ -356,31 +357,49 @@ def save_to_file(self, mot_file):
...
 
         Parameters
         ----------
         solution_type : str
-            Soultion type, which must be ``"EMagnetic"``.
+            Solution type, which must be ``"EMagnetic" or "Thermal"``.
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:390]:
```suggestion
            Solution type, which must be ``"EMagnetic"`` or ``"Thermal"``.
```

**Review by jgsdavies — CHANGES_REQUESTED** [review]

**Review by jgsdavies — APPROVED** [review]
